RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs

```diff
@@ -13,6 +16,7 @@ namespace RepoDb
     public static class DbFieldCache
     {
         private static readonly ConcurrentDictionary<long, DbFieldCollection> cache = new();
+        private static readonly ConcurrentDictionary<IDbHelper, SemaphoreSlim> semaphores = new();
```

Comment:
We can have a single semaphore if we do not care about having simultaneous calls for different DBMSs